packages/vm/src/context.rs

```diff
@@ -316,37 +340,16 @@ mod test {
             let req: QueryRequest<Never> = QueryRequest::Bank(BankQuery::AllBalances {
                 address: HumanAddr::from(INIT_ADDR),
             });
-            querier.raw_query(&to_vec(&req).unwrap())
+            Ok(querier.raw_query(&to_vec(&req).unwrap())?)
         })
         .unwrap()
+        .unwrap()
         .unwrap();
         let balance: AllBalanceResponse = from_binary(&res).unwrap();
 
         assert_eq!(balance.amount, coins(INIT_AMOUNT, INIT_DENOM));
     }
 
-    #[test]
-    fn with_querier_from_context_parse_works() {
-        let mut instance = make_instance();
-        let ctx = instance.context_mut();
-        leave_default_data(ctx);
-        let contract = HumanAddr::from(INIT_ADDR);
-
-        let balance = with_querier_from_context::<S, Q, _, _>(ctx, |querier| {
-            Ok(querier.query_balance(&contract, INIT_DENOM))
-        })
-        .unwrap()
-        .unwrap();
-        assert_eq!(balance.amount, coin(INIT_AMOUNT, INIT_DENOM));
-
-        let balance = with_querier_from_context::<S, Q, _, _>(ctx, |querier| {
-            Ok(querier.query_balance(&contract, "foo"))
-        })
-        .unwrap()
-        .unwrap();
-        assert_eq!(balance.amount, coin(0, "foo"));
-    }
```

Comment:
What happened to this test, is it unnecessary? If I understand correctly, this calls `querier.query_balance` and the test above calls `querier.raw_query`.

Reply:
`query_balance` is irrelevant in the vm layer. It is not used, and never implemented directly, instead it relies on the `raw_query` implementation. The vm only forwards raw queries between the contract and the Go module